classes/local/form/verification_field.php

```diff
@@ -0,0 +1,70 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * MFA Verification code element.
+ *
+ * @package     tool_mfa
+ * @author      Peter Burnett <peterburnett@catalyst-au.net>
+ * @copyright   Catalyst IT
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_mfa\local\form;
+
+
+defined('MOODLE_INTERNAL') || die();
+require_once($CFG->libdir . '/form/text.php');
+
+class verification_field extends \MoodleQuickForm_text {
+
+    public function __construct($attributes=null) {
+        global $PAGE;
+
+        // Force attributes.
+        if (empty($attributes)) {
+            $attributes = [];
+        }
+
+        $attributes['autocomplete'] = 'one-time-code';
+        $attributes['autofocus'] = 'autofocus';
+        $attributes['inputmode'] = 'numeric';
+        $attributes['pattern'] = '[0-9]*';
+        $attributes['style'] = '
```

Comment:
can you add this as a class and move the css to .../styles.css which is magically loaded into the main css file, this way a theme can override it